src/ert/ensemble_evaluator/snapshot.py

```diff
@@ -397,68 +393,41 @@ class ForwardModel(TypedDict, total=False):
     stderr: Optional[str]
 
 
-class RealizationSnapshot(BaseModel):
-    status: Optional[str] = None
-    active: Optional[bool] = None
-    start_time: Optional[datetime] = None
-    end_time: Optional[datetime] = None
-    forward_models: Dict[str, ForwardModel] = {}
-
-
-class SnapshotDict(BaseModel):
-    status: Optional[str] = state.ENSEMBLE_STATE_UNKNOWN
-    reals: Dict[str, RealizationSnapshot] = {}
-    metadata: Dict[str, Any] = {}
+class RealizationSnapshot(TypedDict, total=False):
+    status: Optional[str]
+    active: Optional[bool]
+    start_time: Optional[datetime]
+    end_time: Optional[datetime]
+    forward_models: Dict[str, ForwardModel]
+
+
+def _realization_dict_to_realization_snapshot(
+    realization_data: Dict[str, Any],
+) -> RealizationSnapshot:
+    realization_snapshot = RealizationSnapshot(
+        status=realization_data.get("status"),
+        active=realization_data.get("active"),
+        start_time=realization_data.get("start_time"),
+        end_time=realization_data.get("end_time"),
+    )
+
+    realization_snapshot["forward_models"] = {}
+
+    for fm_id, fm_data in realization_data.get("forward_models", {}).items():
+        realization_snapshot["forward_models"][fm_id] = fm_data
+        if "current_memory_usage" in realization_snapshot["forward_models"][fm_id]:
+            realization_snapshot["forward_models"][fm_id]["current_memory_usage"] = str(
+                realization_snapshot["forward_models"][fm_id]["current_memory_usage"]
```

Comment:
There are tests passing `...memory_usage` as ints, and for some reason we are expecting it to be strings. This caused pydantic to blow up.